numberOfProvinces.js

```diff
@@ -0,0 +1,106 @@
+/*
+n cities
+
+a - b - c
+
+i: n x n matrix
+o: total number of provinces
+
+directly or indirectly
+
+Input: isConnected = [[1,1,0],[1,1,0],[0,0,1]]
+Output: 2
+
+   0 1 2
+0 [1,1,0],
+1 [1,1,0],
+2 [0,0,1]
+
+*/
+
+const { Queue } = require('../utils/queue');
+
+const directions = [
+    [1, 0],
+    [0, 1],
+    [-1, 0],
+    [0, -1],
+];
+
+const CONNECTED = 1;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isBound = (row, col, matrix) => {
+    const rowBound = matrix.length;
```

Comment:
I don't think this name is clear. This variable stores the number of rows, so I'd go with something like `numRows`.